src/hera/workflows/v5/script.py

```diff
@@ -116,7 +119,51 @@ def _build_source(self) -> str:
             assert isinstance(self.source, str)
             return self.source
 
-    def _build_template(self) -> _ModelScriptTemplate:
+    def _build_template(self) -> _ModelTemplate:
```

Comment:
This fixes a bug with the Script template where it was returning a sub-object instead of the actual template.